src/clojurewerkz/elastisch/native/bulk.clj

```diff
@@ -0,0 +1,74 @@
+;; Copyright 2011-2014 Michael S. Klishin, Alex Petrov, and the ClojureWerkz Team
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;     http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+
+(ns clojurewerkz.elastisch.native.bulk
+  (:refer-clojure :exclude [get replace count sort])
+  (:require [clojurewerkz.elastisch.native :as native]
+            [clojure.string :as string]
+            [clojure.set :refer :all]
+            [clojurewerkz.elastisch.native.conversion :as cnv]
+            [clojurewerkz.elastisch.arguments :as ar]
+            [clojurewerkz.elastisch.common.bulk :as common-bulk])
+  (:import clojure.lang.IPersistentMap
+           org.elasticsearch.client.Client
+           org.elasticsearch.action.bulk.BulkRequest
+           org.elasticsearch.action.bulk.BulkRequestBuilder
+           org.elasticsearch.action.bulk.BulkResponse
+           org.elasticsearch.action.index.IndexRequest
+           org.elasticsearch.action.delete.DeleteRequest))
+
+(defmulti add-operation (fn [_ operation] (type operation)))
```

Comment:
You have just re-invented protocols, except a lot less efficiently (because you still do double dispatch). Please switch to [protocols](http://clojure-doc.org/articles/language/polymorphism.html).